Changing Contributing document to reflect process

jeremyf opened the pull request.

jeremyf committed:
Initial commit for proposed CONTRIBUTING doc

This commit includes the proposition that we will be using Git Flow as
our Git workflow procedure.

HYDRASIR-62 #start Working document regarding

JIRA References:
https://jira.duraspace.org/browse/HYDRASIR-62

jeremyf committed:
Adding contributor notes for commit messages

Explaining that you can transition stories via commit messages.

HYDRASIR-62

JIRA References:
https://jira.duraspace.org/browse/HYDRASIR-62

jeremyf committed:
Fleshing out Reviewer/Submitter responsibilities

HYDRASIR-62 #close Fleshing out Reviewer/Submitter responsibilities

[skip ci]

JIRA References:
https://jira.duraspace.org/browse/HYDRASIR-62

jeremyf commented:
Attention: @projecthydra/hydra-curate ; What is missing? I removed a lot about how to make a feature branch. Would that be helpful?

Adding this for review. And sorry @jcoyne but Git Flow is back on the table for discussion.

jeremyf committed:
Adding possible gotchas

hortongn commented:
Your heading "Responsibilities a Reviewer" is missing a word.

This will be a really valuable document for all involved.  It's unfortunate that this process has to be so complex, but it sounds like this is what's needed to end up with a quality product.

jeremyf committed:
Fixing typo

[ci skip][log skip]

JIRA References:
https://jira.duraspace.org/browse/HYDRASIR-62
